geps/gep-1651.md

```diff
@@ -0,0 +1,274 @@
+# GEP-1651: Gateway Reachability
+
+* Issue: [#1651](https://github.com/kubernetes-sigs/gateway-api/issues/1651)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+Allow users to configure a GatewayAddress so that it is only reachable within
+a specific scope (ie. public/cluster local)
+
+## Goals
+
+- Cluster local gateways can be created in a declarative way (`cluster`) and
+  is accessible within a the bounds of the cluster's CNI
+- Define a `public` scope for addresses that are accessible to the internet
+- Implementations can define & support their own scopes
+
+## Non-Goals
+
+- Per-request/route scope
+- Not a lightweight service mesh
+
+## Introduction
+
+One of the early feature requests for Knative was the ability to deploy an
+application using Knative's HTTP routing support, but make it only available
+within the cluster. I want to be able to specify both the "internal"
+(service.namespace.svc) and "external" (service.namespace.example.com).
+Gateways using the same GatewayClass on the cluster, but ensure that the
+"internal" service is only reachable within the cluster. This would greatly
+simplify deployment for users over the instructions we have today.
+
+The reason why we require an "internal" Gateway is so we can take advantage of
+layer 7 load balancing for our internal services.
+
+## API
+
+We propose adding a new `reachability` field under the `spec.infrastructure` stanza of a Gateway.
+
+### Predefined Reachability Values
+
+Implementations MAY implement the following values for 'reachability' and MUST abide by
+their defined semantics.
+
+Value | Semantic
```

Comment:
I personally don't want to see this move to "implementable" until this can address all common use cases, including internal load balancers. I'm fine with this being a follow up from this PR, but not sure how much time I'll have to contribute here.

Reply:
We now have `Private` as per the discussion https://github.com/kubernetes-sigs/gateway-api/pull/1653#discussion_r1183089113